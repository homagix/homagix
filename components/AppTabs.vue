<script setup lang="ts">
type Tab = { id: string } & Record<string, unknown>

defineProps<{
  tabs: Tab[]
  labelField: string
}>()

const selected = defineModel<Tab>()
</script>

<template>
  <span class="tabs">
    <a v-for="tab in tabs" @click="selected = tab" :class="{ selected: tab.id === selected?.id }">{{
      tab[labelField]
    }}</a>
  </span>
</template>

<style lang="scss" scoped>
.tabs {
  display: flex;
  gap: 1rem;

  a {
    padding: 10px 15px;
    font-weight: bold;
    text-align: center;
    cursor: pointer;
    border: none;
    background-color: #f9f9f9;
    color: #333;
    font-size: 16px;
    border-bottom: 2px solid transparent;
    transition: background-color 0.3s, border-bottom 0.3s;

    &:hover {
      background-color: #eaeaea;
    }

    &.selected {
      border-bottom: 2px solid #f78e00;
      color: #f78e00;
      background-color: #fff;
    }
  }
}
</style>
