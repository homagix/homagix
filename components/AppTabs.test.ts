import { describe, it, expect } from "vitest"
import { setup } from "@nuxt/test-utils"
import { mountSuspended } from "@nuxt/test-utils/runtime"
import AppTabs from "./AppTabs.vue"

const tabs = [
  { id: "tab-1", label: "My tab no 1" },
  { id: "tab-2", label: "My tab no 2" },
]
describe("AppTabs", () => {
  setup({ server: false })

  it("renders as expected", async () => {
    const wrapper = await mountSuspended(AppTabs, {
      props: { labelField: "label", tabs, modelValue: tabs[1] },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("reacts on tab switches", async () => {
    const wrapper = await mountSuspended(AppTabs, {
      props: { labelField: "label", tabs, modelValue: tabs[1] },
    })
    expect(wrapper.emitted()).toEqual({})
    await wrapper.find("a").trigger("click")
    expect(wrapper.html()).toMatchSnapshot()
  })
})
