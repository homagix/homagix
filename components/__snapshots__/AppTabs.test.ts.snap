// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppTabs > reacts on tab switches 1`] = `"<span data-v-d9d9d376="" class="tabs"><a data-v-d9d9d376="" class="selected">My tab no 1</a><a data-v-d9d9d376="" class="">My tab no 2</a></span>"`;

exports[`AppTabs > renders as expected 1`] = `"<span data-v-d9d9d376="" class="tabs"><a data-v-d9d9d376="" class="">My tab no 1</a><a data-v-d9d9d376="" class="selected">My tab no 2</a></span>"`;
